python/http_ece/__init__.py

```diff
@@ -86,47 +153,99 @@ def lengthPrefix(key):
         info=nonceinfo,
         backend=default_backend()
     )
-    result = (hkdf_key.derive(secret), hkdf_nonce.derive(secret))
-    return result
+    try:
+        return hkdf_key.derive(secret), hkdf_nonce.derive(secret)
+    except TypeError as ex:
+        print (ex)
+
 
 def iv(base, counter):
     if (counter >> 64) != 0:
-        raise Exception(u"Counter too big")
+        raise ECEException(u"Counter too big")
     (mask,) = struct.unpack("!Q", base[4:])
     return base[:4] + struct.pack("!Q", counter ^ mask)
 
-def decrypt(buffer, salt, key=None, keyid=None, dh=None, rs=4096, authSecret=None, padSize=2):
-    def decryptRecord(key, nonce, counter, buffer):
+
+def decrypt(content, salt, key=None, keyid=None, dh=None, rs=4096,
+            auth_secret=None, pad_size=2, content_type="aesgcm", **kwargs):
```

Comment:
content_type might be a bad name

the default for content_type is unfortunate.  I would prefer to have this default to the new thing.  (My js code does.)  Is it too much to ask that existing users include content_type="aesgcm" in their code?

since pad_size is a legacy argument, I would remove it from here and fall back to your code with kwargs